### PR TITLE
Add priority of constituencies

### LINKED ISSUE #906: Define a "Priority of Constituencies"
The [Web Platform Design Principles](https://www.w3.org/TR/design-principles/) are the guidelines for designing web & browser apis. They define their ["Priority of Constituencies"](https://www.w3.org/TR/design-principles/#priority-of-constituencies)

> If a trade-off needs to be made, always put user needs above all.
>
> Similarly, when beginning to design an API, be sure to understand and document the user need that the API aims to address.
>
>[The internet is for end users](https://tools.ietf.org/html/rfc8890): any change made to the web platform has the potential to [affect vast numbers of people](https://www.w3.org/2001/tag/doc/ethical-web-principles/#allpeople), and may have a profound impact on any person’s life. [[RFC8890]](https://www.w3.org/TR/design-principles/#biblio-rfc8890)
>
> User needs come before the needs of web page authors, which come before the needs of user agent implementors, which come before the needs of specification writers, which come before theoretical purity.
>
>Like all principles, this isn’t absolute. Ease of authoring affects how content reaches users. User agents have to prioritize finite engineering resources, which affects how features reach authors. Specification writers also have finite resources, and theoretical concerns reflect underlying needs of all of these groups.

And since this spec also has to balance the needs of "writers", "operators" and "implementors" I thought it could be valuable to define similar guiding principles to help us make decisions. 

Obviously we do no have to go with their wording (or even their order, even though I agree with it). A good place for it could be in the or next to the [Design](https://github.com/serverlessworkflow/specification/blob/main/dsl.md#design) section of the dsl.

### PR DESCRIPTION
**Pull request by matthias-pichler-warrify**
Signed-off-by: Matthias Pichler <m.pichler@warrify.com>

**Please specify parts of this PR update:**

- [ ] Specification
- [ ] Schema
- [ ] Examples
- [ ] Extensions
- [ ] Use Cases
- [ ] Community
- [ ] CTK
- [x] Other

**Discussion or Issue link**:

Closes #906

**What this PR does**:

Adds a priority of user needs simialr to the w3c

**Additional information:**

### CODE REVIEW

**Review by cdavernas — APPROVED** [review]
LGTM! Thanks man ❤️

File: dsl.md [hunk 1]
```diff
@@ -40,6 +41,23 @@ Serverless computing has gained popularity for its ability to abstract away infr
...
+## Priority of Constituencies
+
+Inspired by the [Priority of Constituencies](https://www.w3.org/TR/2024/NOTE-design-principles-20240718/#priority-of-constituencies) principle from the W3C Design Principles, the Serverless Workflow DSL prioritizes the following constituencies (collectively referred to as "users"):
+
+- Authors: people authoring and reading workflows
+- Operators: people running and operating a runtime implementation of the specification
+- Implementors: people implementing a specification compliant runtime
+- Specifications writers: people working on the specifications of Serverless Workflow
+
+If a trade-off needs to be made, always put author needs above all.
```
Comment (ricardozanini) [inline dsl.md:53]:
```suggestion
If a trade-off needs to be made, always put author's needs above all.
```

File: dsl.md [hunk 2]
```diff
@@ -40,6 +41,23 @@ Serverless computing has gained popularity for its ability to abstract away infr
...
+
+Similarly, when beginning to design an API, be sure to understand and document the user need that the API aims to address.
```
Comment (ricardozanini) [inline dsl.md:55]:
```suggestion
Similarly, when beginning to design an API, be sure to understand and document the user needs that the API aims to address.
```

File: dsl.md [hunk 3]
```diff
@@ -40,6 +41,23 @@ Serverless computing has gained popularity for its ability to abstract away infr
...
+
+Author needs come before the needs of operators, which come before the needs of runtime implementors, which come before the needs of specification writers, which come before theoretical purity.
+
+Like all principles, this isn’t absolute. Ease of operations affects perceived reliability of authors workflows. Implementors have to prioritize finite engineering resources, which affects how features reach authors. Specification writers also have finite resources, and theoretical concerns reflect underlying needs of all of these groups.
```
Comment (ricardozanini) [inline dsl.md:59]:
```suggestion
Like all principles, this isn’t absolute. Ease of operations affects the perceived reliability of authors' workflows. Implementors have to prioritize finite engineering resources, which affects how features reach authors. Specification writers also have finite resources, and theoretical concerns reflect the underlying needs of all of these groups.
```

**Review by ricardozanini — COMMENTED** [review]
Small nits if you don't mind. Many thanks!

**Review by ricardozanini — APPROVED** [review]

**Pull request merged**